### PR TITLE
Make sure to use Xargo.toml directory when resolving path

### LINKED ISSUE #259: Canonicalize issue with Xargo.toml in parent directory
Since #254, xargo can search for Xargo.toml in parent directories.
However relative paths in the Xargo.toml aren't canonicalized using the correct root directory.

I made a repository to trigger the bug https://github.com/Thog/xargo_canonicalize_issue

This was tested on commit 08bf79ccf5d95ba1d5b71943145953e02eed0fa0

### PR DESCRIPTION
**Pull request by Thog**
This fix #259

### CODE REVIEW

**Comment by RalfJung** [comment]
In https://github.com/japaric/xargo/issues/259 you talk about the path not being "canonicalized" correctly. I have no idea how that's an issue. Could you explain a bit more?

Is there any way the existing test for parent path Xargo.toml could be adjusted to cover this case (so that the bug does not come back)?

File: src/sysroot.rs [hunk 1]
```diff
@@ -231,9 +231,11 @@ pub fn update(
     verbose: bool,
 ) -> Result<()> {
     let ctoml = cargo::toml(root)?;
-    let xtoml = xargo::toml(root)?;
+    let (xtoml_parent, xtoml) = xargo::toml(root)?;
 
-    let blueprint = Blueprint::from(xtoml.as_ref(), cmode.triple(), root, &src)?;
+    let base_path = xtoml_parent.unwrap_or_else(|| root.path().to_path_buf());
```
Comment (RalfJung) [inline src/sysroot.rs]: Seems a bit wasteful to make a copy here with `to_path_buf` if we only need an `&Path` below anyway. Can that be avoided?

**Comment by jethrogb** [comment]
> In #259 you talk about the path not being "canonicalized" correctly. I have no idea how that's an issue. Could you explain a bit more?

@RalfJung When you use a relative path in `Xargo.toml`, you expect that to be resolved relative to the location of `Xargo.toml`. #259 claims that isn't done properly.

**Comment by RalfJung** [comment]
Ah, yes that makes sense.

Is there any way we could have a test for this (e.g. adjusting the existing test)? (If that's too difficult we can proceed without, but I am worried that this could easily break again.)

**Comment by RalfJung** [comment]
Thanks, the code looks good. Did you think about tests?

Also I think there should be a comment somewhere explaining what is going on. Clearly something is subtle here as the current code works in most but not all cases; any time the code does something subtle there should be comments pointing out that this is subtle and what it is that is being done.

**Comment by RalfJung** [comment]
That's awesome. :) Let's see what CI says.

bors r+

**Pull request merged**